internal/mcpproxy/authorization.go

```diff
@@ -241,6 +249,89 @@ func scopesSatisfied(have sets.Set[string], required []string) bool {
 	return true
 }
 
+func claimsSatisfied(claims jwt.MapClaims, required []filterapi.JWTClaim) bool {
+	if len(required) == 0 {
+		return true
+	}
+
+	for _, claim := range required {
+		value, ok := lookupClaim(claims, claim.Name)
+		if !ok {
+			return false
+		}
+
+		valueType := claim.ValueType
+		if valueType == "" {
```

Comment:
This can't happen, given how we're populating the internal api values? We already default to String?